daemon/algod/api/server/v2/dryrun.go

```diff
@@ -119,48 +119,31 @@ type dryrunDebugReceiver struct {
 }
 
 func (ddr *dryrunDebugReceiver) updateScratch() {
-	any := false
 	maxActive := -1
 	lasti := len(ddr.history) - 1
 
 	if ddr.history[lasti].Scratch == nil {
 		return
 	}
 
+	ddr.scratchActive = make([]bool, 256)
 	for i, sv := range *ddr.history[lasti].Scratch {
 		if sv.Type != uint64(basics.TealUintType) || sv.Uint != 0 {
-			any = true
+			ddr.scratchActive[i] = true
 			maxActive = i
 		}
 	}
 
-	if any {
-		if ddr.scratchActive == nil {
-			ddr.scratchActive = make([]bool, maxActive+1, 256)
-		}
-		for i := len(ddr.scratchActive); i <= maxActive; i++ {
-			sv := (*ddr.history[lasti].Scratch)[i]
-			active := sv.Type != uint64(basics.TealUintType) || sv.Uint != 0
-			ddr.scratchActive = append(ddr.scratchActive, active)
-		}
-	} else {
-		if ddr.scratchActive != nil {
-			*ddr.history[lasti].Scratch = (*ddr.history[lasti].Scratch)[:len(ddr.scratchActive)]
-		} else {
-			ddr.history[lasti].Scratch = nil
-			return
-		}
-	}
-
-	scratchlen := maxActive + 1
-	if len(ddr.scratchActive) > scratchlen {
-		scratchlen = len(ddr.scratchActive)
+	if maxActive == -1 {
+		ddr.history[lasti].Scratch = nil
+		return
 	}
 
-	*ddr.history[lasti].Scratch = (*ddr.history[lasti].Scratch)[:scratchlen]
+	*ddr.history[lasti].Scratch = (*ddr.history[lasti].Scratch)[:maxActive+1]
 	for i := range *ddr.history[lasti].Scratch {
+		// TODO: Still not sure why we need this?
 		if !ddr.scratchActive[i] {
-			(*ddr.history[lasti].Scratch)[i].Type = 0
+			(*ddr.history[lasti].Scratch)[i] = generated.TealValue{}
```

Comment:
If we're zero'ing the type, presumably we dont want _anything_ in the uint/bytes either so set this to empty struct

Reply:
I suspect we do it to ensure that nothing appears in the response. Though I'm unclear why it would have been previously populated. Is that your question as well? I'm not sure how `ddr.history[lasti]` is set.